geo_autoRIFT/autoRIFT/autoRIFT.py

```diff
@@ -97,6 +99,78 @@ def _wallis_filter_fill(image, filter_width, std_cutoff):
     return image, zero_mask
 
 
+def _find_largest_region(arr):
+    binary_arr = np.zeros(arr.shape)
+    binary_arr[arr != 0] = 1
+    label_arr, nb_labels = ndimage.label(binary_arr)
+    sizes = ndimage.sum(binary_arr, label_arr, range(nb_labels + 1))
+    max_label = sizes.argmax()
+    label_arr[label_arr != max_label] = 0
+    return label_arr
+
+
+def _fft_filter(Ix, valid_domain, power_threshold=500):
+    import numpy.fft as fft
+    m, n = valid_domain.shape
+    center_m = int(np.floor(m / 2))
+    center_n = int(np.floor(n / 2))
+
+    single_region = _find_largest_region(Ix)
+    single_region = np.uint8(single_region * 255)
+    contours, hierarchy = cv2.findContours(single_region, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
+    if hierarchy.shape[1] > 1:
+        raise ValueError(f'{hierarchy.shape[1]} external objects founds, only expecting 1.')
+    contour = contours[0]
+    moment = cv2.moments(contour)
+
+    centroid_m = int(np.floor(moment['m01'] / moment['m00']))
+    centroid_n = int(np.floor(moment['m10'] / moment['m00']))
+    rectangle = cv2.minAreaRect(contour)
+    angle = rectangle[2]
+
+    filter_base = np.full((m, n), False)
+    filter_base[center_m - 70:center_m + 70, :] = 1
+    filter_base[:, center_n - 100:center_n + 100] = 0
+
+    filter_a = ndimage.rotate(filter_base, -angle, reshape=False)
+    filter_b = ndimage.rotate(filter_base, 90 - angle, reshape=False)
+
+    ctr_shift = [centroid_m - center_m, centroid_n - center_n]
+
+    translate_matrix = np.array([(1, 0, ctr_shift[0]), (0, 1, ctr_shift[1]), (0, 0, 1)])
+    filter_a = ndimage.affine_transform(filter_a, matrix=translate_matrix)
```

Comment:
I think replacing L 140-142 with
    filter_a = ndimage.shift(filter_a, ctr_shift)
    filter_b = ndimage.shift(filter_b, ctr_shift)
might improve the banding issues we're observing.

Reply:
We are now using CV2 to perform this translation, which appears to be working correctly.